packages/next/src/server/async-storage/static-generation-async-storage-wrapper.ts

```diff
@@ -31,7 +25,18 @@ export type StaticGenerationContext = {
      */
     // TODO: remove this when we resolve accessing the store outside the execution context
     store?: StaticGenerationStore
-  }
+  } & Pick<
+    // Pull some properties from RenderOptsPartial so that the docs are also
+    // mirrored.
+    RenderOptsPartial,
+    | 'originalPathname'
+    | 'supportsDynamicHTML'
+    | 'isRevalidate'
+    | 'isBot'
+    | 'nextExport'
+    | 'isDraftMode'
+    | 'isDebugPPRSkeleton'
+  >
```

Comment:
Maybe something to follow up on later but having to repeat these values just so they can be exposed in async context is probably not good long term. We should maybe just make renderOpts available through a generalization of the staticGenerationContext (it's not really about staticGeneration only anymore anyway)

Reply:
I'd argue in favour of narrowing the types here. The biggest problem with `RenderOpts` is that it's overloaded, and holds way too many fields. Scoping down the fields pulled from here is a small step towards breaking the type up into smaller parts.